corehq/motech/fhir/models.py

```diff
@@ -334,3 +337,103 @@ def deepmerge(a, b):
         return a
     else:
         return b
+
+
+class SmartConfiguration(JsonObject):
+    """
+    The /.well-known/smart-configuration.json response for a domain.
+    http://hl7.org/fhir/smart-app-launch/conformance/index.html#using-well-known
+    """
+    authorization_endpoint = StringProperty(required=True)
+    token_endpoint = StringProperty(required=True)
+    token_endpoint_auth_methods = ListProperty(
+        StringProperty(choices=["client_secret_post", "client_secret_basic"]), exclude_if_none=True
+    )
+    registration_endpoint = StringProperty(exclude_if_none=True)
+    scopes_supported = ListProperty(StringProperty, exclude_if_none=True)
+    response_types_supported = StringProperty(exclude_if_none=True)
+    management_endpoint = StringProperty(exclude_if_none=True)
+    revokation_endpoint = StringProperty(exclude_if_none=True)
+    capabilities = ListProperty(StringProperty, exclude_if_none=True)
+
+
+class _Coding(JsonObject):
+    code = StringProperty(exclude_if_none=True)
+    system = StringProperty(exclude_if_none=True)
+
+
+class _CodableConcept(JsonObject):
+    coding = ListProperty(_Coding)
+
+
+class _ExtensionURI(JsonObject):
+    url = StringProperty()
+    valueUri = StringProperty()
+
+
+class _Extension(JsonObject):
+    url = StringProperty()
+    extension = ListProperty(_ExtensionURI)
+
+
+class _RestSecurity(JsonObject):
+    service = ListProperty(
+        _CodableConcept,
+        default=[
+            _CodableConcept(
+                coding=[_Coding(system="http://hl7.org/fhir/restful-security-service", code='SMART-on-FHIR')]
+            )
+        ]
+    )
+    extension = ListProperty(_Extension)
+
+
+class _MetadataRest(JsonObject):
+    mode = StringProperty(required=True, default='server')
+    security = ObjectProperty(_RestSecurity, default=_RestSecurity())
+
+
+class SmartCapabilityStatement(JsonObject):
+    """
+    The /metadata response for a domain which contains the FHIR CapabilityStatement
+    http://hl7.org/fhir/capabilitystatement.html
+    """
+    format = ListProperty(StringProperty(), required=True, default=['json'])
+    status = StringProperty(required=True, default='active')
+    kind = StringProperty(required=True, default='instance')
+    fhirVersion = StringProperty(required=True, default=FHIR_VERSION_4_0_1)
+    rest = ListProperty(_MetadataRest, required=True, default=[_MetadataRest()])
+    date = StringProperty(required=True, default='2021-03-23')
```

Comment:
why this date?

Reply:
This is the "date (and optionally time) when the capability statement was published. The date must change when the business version changes and it must change if the status code changes. In addition, it should change when the substantive content of the capability statement changes." https://www.hl7.org/fhir/capabilitystatement-definitions.html#CapabilityStatement.date
I will make this clearer in the code.

Reply:
https://github.com/dimagi/commcare-hq/pull/29453/commits/b5e2b13297b36f1fcdaf4ec9be4ec714a6a44efc